File: Eisenhower.java
// Eisenhower.java
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
import com.fasterxml.jackson.core.type.TypeReference;

import java.io.File;
import java.io.IOException;
import java.time.Instant;
import java.util.*;
import java.util.stream.Collectors;

class Task {
    public String id;
    public String title;
    public String description;
    public int quadrant;
    public String created;
    public boolean completed;

    public Task() {}
    public Task(String title, String description, int quadrant) {
        this.id = UUID.randomUUID().toString().substring(0,8);
        this.title = title;
        this.description = description;
        this.quadrant = quadrant;
        this.created = Instant.now().toString();
        this.completed = false;
    }
}

public class Eisenhower {
    private List<Task> tasks = new ArrayList<>();
    private final File dataFile = new File("tasks.json");
    private final ObjectMapper mapper = new ObjectMapper()
            .enable(SerializationFeature.INDENT_OUTPUT);

    public Eisenhower() {
        load();
    }

    private void load() {
        if (!dataFile.exists()) return;
        try {
            tasks = mapper.readValue(dataFile, new TypeReference<List<Task>>(){});
        } catch (IOException e) {
            System.err.println("Error loading tasks: " + e.getMessage());
        }
    }

    private void save() {
        try {
            mapper.writeValue(dataFile, tasks);
        } catch (IOException e) {
            System.err.println("Error saving tasks: " + e.getMessage());
        }
    }

    public void addTask(String title, String desc, int quadrant) {
        Task t = new Task(title, desc, quadrant);
        tasks.add(t);
        save();
        System.out.printf("Task added: %s (ID: %s)%n", t.title, t.id);
    }

    public void deleteTask(String id) {
        boolean removed = tasks.removeIf(t -> t.id.equals(id));
        if (removed) {
            save();
            System.out.printf("Task %s deleted.%n", id);
        } else {
            System.out.printf("Task %s not found.%n", id);
        }
    }

    public void moveTask(String id, int newQuadrant) {
        for (Task t : tasks) {
            if (t.id.equals(id)) {
                t.quadrant = newQuadrant;
                save();
                System.out.printf("Task %s moved to quadrant %d.%n", id, newQuadrant);
                return;
            }
        }
        System.out.printf("Task %s not found.%n", id);
    }

    public void listTasks(Integer quadrant) {
        List<Task> filtered = tasks.stream()
                .filter(t -> quadrant == null || t.quadrant == quadrant)
                .collect(Collectors.toList());
        if (filtered.isEmpty()) {
            System.out.println("No tasks found.");
            return;
        }
        String[] labels = {"Urgent & Important", "Not Urgent but Important",
                "Urgent but Not Important", "Not Urgent & Not Important"};
        String[] colors = {"\033[31m", "\033[32m", "\033[33m", "\033[37m"};
        System.out.println("\nEisenhower Matrix");
        System.out.println("=".repeat(60));
        for (int q = 1; q <= 4; q++) {
            System.out.println(colors[q-1] + labels[q-1] + "\033[0m");
            List<Task> qTasks = filtered.stream().filter(t -> t.quadrant == q).collect(Collectors.toList());
            if (qTasks.isEmpty()) {
                System.out.println("  (empty)");
            } else {
                for (Task t : qTasks) {
                    System.out.printf("  [%s] %s - %s%n", t.id, t.title,
                            t.description.isEmpty() ? "No description" : t.description);
                }
            }
            if (q < 4) System.out.println("-".repeat(40));
        }
        System.out.println("=".repeat(60));
    }

    public void stats() {
        int[] counts = new int[5];
        for (Task t : tasks) counts[t.quadrant]++;
        String[] labels = {"Urgent & Important", "Not Urgent but Important",
                "Urgent but Not Important", "Not Urgent & Not Important"};
        System.out.println("Eisenhower Statistics");
        for (int q = 1; q <= 4; q++) {
            System.out.printf("%s: %d tasks%n", labels[q-1], counts[q]);
        }
    }

    public static void main(String[] args) {
        if (args.length < 1) {
            System.out.println("Usage: java Eisenhower <command> [options]");
            return;
        }
        Eisenhower app = new Eisenhower();
        String cmd = args[0];
        switch (cmd) {
            case "add": {
                String title = "", desc = "";
                int quadrant = 1;
                for (int i = 1; i < args.length; i++) {
                    if (args[i].equals("-t") && i+1 < args.length) title = args[++i];
                    if (args[i].equals("-d") && i+1 < args.length) desc = args[++i];
                    if (args[i].equals("-q") && i+1 < args.length) quadrant = Integer.parseInt(args[++i]);
                }
                app.addTask(title, desc, quadrant);
                break;
            }
            case "list": {
                Integer quadrant = null;
                for (int i = 1; i < args.length; i++) {
                    if (args[i].equals("-q") && i+1 < args.length) quadrant = Integer.parseInt(args[++i]);
                }
                app.listTasks(quadrant);
                break;
            }
            case "move": {
                String id = "";
                int q = 0;
                for (int i = 1; i < args.length; i++) {
                    if (args[i].equals("--id") && i+1 < args.length) id = args[++i];
                    if (args[i].equals("-q") && i+1 < args.length) q = Integer.parseInt(args[++i]);
                }
                app.moveTask(id, q);
                break;
            }
            case "delete": {
                String id = "";
                for (int i = 1; i < args.length; i++) {
                    if (args[i].equals("--id") && i+1 < args.length) id = args[++i];
                }
                app.deleteTask(id);
                break;
            }
            case "stats":
                app.stats();
                break;
            default:
                System.out.println("Unknown command.");
        }
    }
}

File: eisenhower.go
// eisenhower.go
package main

import (
	"bufio"
	"encoding/json"
	"fmt"
	"os"
	"strconv"
	"strings"
	"time"
	"github.com/google/uuid"
)

type Task struct {
	ID          string    `json:"id"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	Quadrant    int       `json:"quadrant"`
	Created     time.Time `json:"created"`
	Completed   bool      `json:"completed"`
}

type App struct {
	Tasks []Task
	file  string
}

func NewApp(file string) *App {
	app := &App{file: file}
	app.load()
	return app
}

func (a *App) load() {
	data, err := os.ReadFile(a.file)
	if err != nil {
		return
	}
	json.Unmarshal(data, &a.Tasks)
}

func (a *App) save() {
	data, _ := json.MarshalIndent(a.Tasks, "", "  ")
	os.WriteFile(a.file, data, 0644)
}

func (a *App) AddTask(title, desc string, quadrant int) {
	task := Task{
		ID:          uuid.New().String()[:8],
		Title:       title,
		Description: desc,
		Quadrant:    quadrant,
		Created:     time.Now(),
	}
	a.Tasks = append(a.Tasks, task)
	a.save()
	fmt.Printf("Task added: %s (ID: %s)\n", task.Title, task.ID)
}

func (a *App) DeleteTask(id string) {
	for i, t := range a.Tasks {
		if t.ID == id {
			a.Tasks = append(a.Tasks[:i], a.Tasks[i+1:]...)
			a.save()
			fmt.Printf("Task %s deleted.\n", id)
			return
		}
	}
	fmt.Printf("Task %s not found.\n", id)
}

func (a *App) MoveTask(id string, newQuadrant int) {
	for i, t := range a.Tasks {
		if t.ID == id {
			a.Tasks[i].Quadrant = newQuadrant
			a.save()
			fmt.Printf("Task %s moved to quadrant %d.\n", id, newQuadrant)
			return
		}
	}
	fmt.Printf("Task %s not found.\n", id)
}

func (a *App) ListTasks(quadrant *int) {
	filtered := []Task{}
	for _, t := range a.Tasks {
		if quadrant == nil || t.Quadrant == *quadrant {
			filtered = append(filtered, t)
		}
	}
	if len(filtered) == 0 {
		fmt.Println("No tasks found.")
		return
	}
	labels := map[int]string{
		1: "Urgent & Important",
		2: "Not Urgent but Important",
		3: "Urgent but Not Important",
		4: "Not Urgent & Not Important",
	}
	fmt.Println("\nEisenhower Matrix")
	fmt.Println(strings.Repeat("=", 60))
	for q := 1; q <= 4; q++ {
		fmt.Printf("\033[%dm%s\033[0m\n", 31+q-1, labels[q])
		has := false
		for _, t := range filtered {
			if t.Quadrant == q {
				fmt.Printf("  [%s] %s - %s\n", t.ID, t.Title, t.Description)
				has = true
			}
		}
		if !has {
			fmt.Println("  (empty)")
		}
		if q < 4 {
			fmt.Println(strings.Repeat("-", 40))
		}
	}
	fmt.Println(strings.Repeat("=", 60))
}

func (a *App) Stats() {
	counts := map[int]int{1:0,2:0,3:0,4:0}
	for _, t := range a.Tasks {
		counts[t.Quadrant]++
	}
	labels := []string{"Urgent & Important", "Not Urgent but Important", "Urgent but Not Important", "Not Urgent & Not Important"}
	fmt.Println("Eisenhower Statistics")
	for i, label := range labels {
		q := i + 1
		fmt.Printf("%s: %d tasks\n", label, counts[q])
	}
}

func main() {
	if len(os.Args) < 2 {
		fmt.Println("Usage: eisenhower <command> [options]")
		return
	}
	app := NewApp("tasks.json")
	cmd := os.Args[1]
	switch cmd {
	case "add":
		if len(os.Args) < 4 {
			fmt.Println("add requires -t title [-d desc] [-q quadrant]")
			return
		}
		title, desc := "", ""
		quadrant := 1
		for i := 2; i < len(os.Args); i++ {
			switch os.Args[i] {
			case "-t": title = os.Args[i+1]; i++
			case "-d": desc = os.Args[i+1]; i++
			case "-q": quadrant, _ = strconv.Atoi(os.Args[i+1]); i++
			}
		}
		app.AddTask(title, desc, quadrant)
	case "list":
		quadrant := -1
		for i := 2; i < len(os.Args); i++ {
			if os.Args[i] == "-q" {
				quadrant, _ = strconv.Atoi(os.Args[i+1]); i++
			}
		}
		var qPtr *int
		if quadrant != -1 {
			qPtr = &quadrant
		}
		app.ListTasks(qPtr)
	case "move":
		id, q := "", 0
		for i := 2; i < len(os.Args); i++ {
			if os.Args[i] == "--id" { id = os.Args[i+1]; i++ }
			if os.Args[i] == "-q" { q, _ = strconv.Atoi(os.Args[i+1]); i++ }
		}
		app.MoveTask(id, q)
	case "delete":
		id := ""
		for i := 2; i < len(os.Args); i++ {
			if os.Args[i] == "--id" { id = os.Args[i+1]; i++ }
		}
		app.DeleteTask(id)
	case "stats":
		app.Stats()
	default:
		fmt.Println("Unknown command")
	}
}
